Makefile.shared

```diff
@@ -311,3 +311,30 @@ partialclean::
 	rm -f driver/compdynlink.mli
 	rm -f driver/compdynlink.mlopt
 
+# The native toplevel
+
+compilerlibs/ocamlopttoplevel.cmxa: $(OPTTOPLEVEL:.cmo=.cmx)
+	$(CAMLOPT) -a -o $@ $(OPTTOPLEVEL:.cmo=.cmx)
+partialclean::
+	rm -f compilerlibs/ocamlopttoplevel.cmxa
+
+# When the native toplevel executable has an extension (e.g. ".exe"),
+# provide a phony 'ocamlnat' synonym
+
+ifneq ($(EXE),)
+.PHYNY: ocamlnat
```

Comment:
Alain Frisch (2016/07/31 06:30 -0700):
> > @@ -311,3 +311,30 @@ partialclean::
> >  	rm -f driver/compdynlink.mli
> >  	rm -f driver/compdynlink.mlopt
> >  
> > +# The native toplevel
> > +
> > +compilerlibs/ocamlopttoplevel.cmxa: $(OPTTOPLEVEL:.cmo=.cmx)
> > +	$(CAMLOPT) -a -o $@ $(OPTTOPLEVEL:.cmo=.cmx)
> > +partialclean::
> > +	rm -f compilerlibs/ocamlopttoplevel.cmxa
> > +
> > +# When the native toplevel executable has an extension (e.g. ".exe"),
> > +# provide a phony 'ocamlnat' synonym
> > +
> > +ifneq ($(EXE),)
> > +.PHYNY: ocamlnat
> 
> Why do we need this synonym for ocamlnat (and not other tools)?

Well I wanted to preserve the ability to do

  make ocamlnat

even under Windows

But I don't mind to remove it if it's something people dislike.